### DISCUSSION TITLE
Feature Request: distinguish required statuses from optional ones in the context of a pull request

### DISCUSSION THREAD

**Discussion by mrpinsky** [community/community#35102] [Pull Requests] [answered]
An average, non-admin contributor to a repository can't see branch protection rules, and so doesn't know which statuses are required and which are optional. The only feedback available is whether the "Merge" button is disabled, but if there are multiple pending or failed/errored statuses, the contributor doesn't know which ones to focus on. This becomes a problem in the case of flaky or experimental statuses.

It would be very helpful if GitHub added a badge to statuses when they display in the context of a PR that indicates whether they are required by branch protection or are optional.

A hacky workaround is to put this information in the description (e.g., `[optional] an optional status`), but this is harder to maintain, and harder or not possible if the status comes from an external service.

**Comment by mrpinsky** [comment, marked as answer]
Never mind, I just realized that there is a `required` badge :doh: :back_away:

**Reply by 508511078** [reply]
U f
